### DISCUSSION TITLE
Renovate not updating pnpm-lock.yaml "Command failed: install-tool node"

### DISCUSSION THREAD

**Discussion by Zemnmez** [renovatebot/renovate#24046] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

action v39.0.1

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Renovate is no longer updating `pnpm-lock.yaml` for any of its upgrades. I attach debug logs that look fishy.

I can't provide a minimal replication case but my repo is public: https://github.com/zemnmez/monorepo

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: lock file error (repository=Zemnmez/monorepo, branch=renovate/node-20.x)
       "cmd": "pnpm",
       "err": {
         "cmd": "/bin/sh -c install-tool node v18.17.1",
         "stderr": "",
         "stdout": "v18.17.1\n[22:46:58.780] INFO (127): Installing tool node v18.17.1...\n[22:46:58.785] INFO (127): tool already installed\n    tool: \"node\"\n9.6.7\n0.18.0\n[22:46:59.196] FATAL (127): EACCES: permission denied, scandir '/tmp/snap-private-tmp'\n    err: {\n      \"type\": \"Error\",\n      \"message\": \"EACCES: permission denied, scandir '/tmp/snap-private-tmp'\",\n      \"stack\":\n          Error: EACCES: permission denied, scandir '/tmp/snap-private-tmp'\n      \"errno\": -13,\n      \"code\": \"EACCES\",\n      \"syscall\": \"scandir\",\n      \"path\": \"/tmp/snap-private-tmp\"\n    }\n[22:46:59.349] INFO (127): Installed tool node with errors in 569ms.\n",
         "options": {
           "cwd": "/tmp/renovate/repos/github/Zemnmez/monorepo",
           "encoding": "utf-8",
           "env": {
             "NPM_CONFIG_CACHE": "/tmp/renovate/cache/others/npm",
             "npm_config_store": "/tmp/renovate/cache/others/pnpm",
             "HOME": "/home/ubuntu",
             "PATH": "/home/ubuntu/.cargo/bin:/home/ubuntu/.local/bin:/go/bin:/home/ubuntu/bin:/home/ubuntu/.npm-global/bin:/home/ubuntu/.cargo/bin:/home/ubuntu/.local/bin:/go/bin:/home/ubuntu/bin:/home/ubuntu/.npm-global/bin:/home/ubuntu/.cargo/bin:/home/ubuntu/.local/bin:/go/bin:/home/ubuntu/bin:/home/ubuntu/.npm-global/bin:/home/ubuntu/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
             "LC_ALL": "C.UTF-8",
             "LANG": "C.UTF-8",
             "CONTAINERBASE_CACHE_DIR": "/tmp/renovate/cache/containerbase"
           },
           "maxBuffer": 10485760,
           "timeout": 900000
         },
         "exitCode": 1,
         "name": "ExecError",
         "message": "Command failed: install-tool node v18.17.1\n",
         "stack": "ExecError: Command failed: install-tool node v18.17.1\n\n    at ChildProcess.<anonymous> (/opt/containerbase/tools/renovate/36.57.3/node_modules/renovate/lib/util/exec/common.ts:99:11)\n    at ChildProcess.emit (node:events:526:35)\n    at ChildProcess.emit (node:domain:489:12)\n    at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)"
       },
       "stdout": undefined,
       "stderr": undefined,
       "type": "pnpm"
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by klippx** [comment]
I get the same, it must be a bad docker image has been pushed

All my renovate PRs are failing with

### ⚠ Artifact update problem

Renovate failed to update an artifact related to this branch. You probably do not want to merge this PR as-is.

♻ Renovate will retry this branch, including artifacts, only when one of the following happens:

 - any of the package files in this branch needs updating, or 
 - the branch becomes conflicted, or
 - you click the rebase/retry checkbox if found above, or
 - you rename this PR's title to start with "rebase!" to trigger it manually

The artifact failure details are included below:

##### File name: yarn.lock

```
v18.17.1
[06:05:46.986] INFO (428): Installing tool node v18.17.1...
[06:05:46.992] INFO (428): tool already installed
    tool: "node"
9.6.7
0.18.0
[06:05:47.450] FATAL (428): EACCES: permission denied, scandir '/tmp/snap-private-tmp'
    err: {
      "type": "Error",
      "message": "EACCES: permission denied, scandir '/tmp/snap-private-tmp'",
      "stack":
          Error: EACCES: permission denied, scandir '/tmp/snap-private-tmp'
      "errno": -13,
      "code": "EACCES",
      "syscall": "scandir",
      "path": "/tmp/snap-private-tmp"
    }
[06:05:47.547] INFO (428): Installed tool node with errors in 561ms.

```

**Reply by viceice** [reply]
Can you set `CONTAINERBASE_LOG_LEVEL=debug` as custom env var[^1]? So we see more detailed logs

[^1]: https://docs.renovatebot.com/self-hosted-configuration/#customenvvariables

**Reply by rarkins** [reply]
Can self hosted users experiment with configuring the earlier sidecar image?

**Reply by klippx** [reply]
I added this to my config but no extra debug logs were shown in a fresh run? Did I misunderstand the usage?

```
  "customEnvVariables": {
    "CONTAINERBASE_LOG_LEVEL": "debug"
  }
```

**Reply by klippx** [reply]
Here is LOG_LEVEL=debug anyway

[1_renovate.txt](https://github.com/renovatebot/renovate/files/12426849/1_renovate.txt)

**Comment by klippx** [comment, marked as answer]
This is probably a duplicate of an active discussion here https://github.com/renovatebot/renovate/discussions/24043

**Reply by viceice** [reply]
yes it's a duplicate

**Comment by jonesbusy** [comment]
Same issue here. 'ENOENT' with at least npm and helm

**Reply by viceice** [reply]
- https://github.com/renovatebot/renovate/discussions/24043#discussioncomment-6811716

**Reply by jonesbusy** [reply]
Thanks, was following this thread. Updated to 36.57.10 and no more issues
